db/db_impl/db_impl_write.cc

```diff
@@ -1801,7 +1804,24 @@ Status DBImpl::SwitchMemtable(ColumnFamilyData* cfd, WriteContext* context) {
   }
   if (s.ok()) {
     SequenceNumber seq = versions_->LastSequence();
-    new_mem = cfd->ConstructNewMemtable(mutable_cf_options, seq);
+    std::unique_ptr<uint32_t> new_filterbits;
+
+    if (mutable_cf_options.memtable_self_tuning_bloom) {
+      const uint64_t pastentries = cfd->mem()->BFUniqueEntryEstimate();
+      new_filterbits.reset(new uint32_t);
+      // Calculate new BF size by assuming the number of unique keys
+      // will be similar to the mutable memtable being switched,
+      // (for 6 probes (hash functions) and a target FPR of 1%)
+      // m = - n_approx * ln(0.01)/[ln(2)**2]
```

Comment:
I don't think we need to be fancy here. The cache locality makes FP rates slightly off anyway. We can just say for 1% FP rate, 10 bits per key and 6 probes is known to work well. You can wiki link to https://github.com/facebook/rocksdb/wiki/RocksDB-Bloom-Filter#the-math